ring-core/src/ring/middleware/file.clj

```diff
@@ -40,15 +40,30 @@
   Accepts the following options:
 
   :index-files?    - look for index.* files in directories, defaults to true
-  :allow-symlinks? - serve files through symbolic links, defaults to false"
+  :allow-symlinks? - serve files through symbolic links, defaults to false
+  :prefer-handler? - give preference to handler to process the file. 
+                     if the handler successfully processed the file, do not serve
+                     static content"
   ([handler root-path]
    (wrap-file handler root-path {}))
   ([handler root-path options]
    (ensure-dir root-path)
    (fn
      ([request]
-      (or (file-request request root-path options) (handler request)))
+      (if (:prefer-handler? options)
+        (let [response (handler request)]
+          (if (not= 404 (:status response))
```

Comment:
Prefer `(= 404 (:status response))` since there's no need for a `not` when we can just reorder the clauses of the `if`.